bpfd-api/src/config.rs

```diff
@@ -4,13 +4,18 @@
 use std::{collections::HashMap, fs, path::Path};
 
 use aya::programs::XdpFlags;
-use bpfd_api::util::directories::*;
 use log::{error, warn};
 use serde::{Deserialize, Serialize};
 
+use crate::util::directories::*;
+
 #[derive(Debug, Deserialize, Default)]
 pub struct Config {
     pub tls: TlsConfig,
+    #[serde(default)]
+    pub bpfctl: BpfctlTlsConfig,
```

Comment:
Can we make this more generic and get rid of the dedicated `BpfdAgentTlsConfig`, `BpfctlTlsConfig` types? I'm assuming we generally just need a `TlsConfig` object for `bpfd` itself and then any and all `bpfd-clients`? and if we don't need a field (like `ca_cert`) then we can just make it an `Option` 

It seems like the main reason for this was for implementing different default functions but that feels clunky to include in the API

Reply:
Have one `TlsConfig` type with bpfd and bpfd-client file paths. All fields are optional in the file, and if not provided will default to a value that makes sense for running on the host. K8s will need to set to filenames used by certManager or equivalent.